### DISCUSSION TITLE
how use HTTP/2

### DISCUSSION THREAD

**Discussion by hktalent** [projectdiscovery/nuclei#4183] [Q&A]
@ehsandeep 
Hello, my friend, I saw that the templates of nuclei are all http/1.1. I wonder if there is a way to dynamically upgrade to http/2 according to the target situation? As you know, http2 is much faster than http 1.1.

**Comment by hktalent** [comment]
https://github.com/search?q=repo%3Aprojectdiscovery%2Fnuclei+HTTP%2F2&type=issues

use
```
-debug-req -fh2
```
but same 
```
GET /npm/vue@2.6.10/dist/vue.min.js?exp=2Vo9OvDpjGn5u69WOSS8A2GCmC0 HTTP/1.1
```

**Reply by hktalent** [reply]
"""
Note: go net/http package always dump requests in HTTP/1.1 format even in case of H2 connection upgrade since the H2 protocol is binary and can't be represented in ASCII. We need to ensure that the connection upgrade happens (probably by hooking Trace callbacks since H2 requires TLS)
"""

**Comment by MetzinAround** [comment]
Hi @hktalent, 

I noticed you seem to have resolved this on your own, but I wanted to let you know you don't have to tag Sandeep in each issue/discussion. We will be able to help you without the extra noise tagging can sometimes cause. Thank you!

**Reply by hktalent** [reply]
ok
